### PR TITLE
feature(ui-ux): refetch hot wallet balance and transfer guard

### LINKED ISSUE #758: UI - Refectch the hot wallet balance on button click
#### What UI/UX improvements would you like to be implemented:
- The hot wallet balance is only fetched on load of the screen. Need to refetch the hot wallet balance and validate it against the input amount after clicking "Review Transaction" to ensure that it is validating.

#### Why is this needed:
- If the user stays in the website for too long, the hot wallet balance will be outdated which could lead to stuck txn

#### Which component/screens does this concern?:
In Bridge form, refetch the hot wallet balance on click of the button and 
``` const { getBalance } = useCheckBalance();```

### PR DESCRIPTION
**Pull request by G9000**
#### What this PR does / why we need it:
To ensure balance value are up to date when verify transfer is clicked and prevent user to proceed if the balance is insufficient.

Added loading state to the button when fetching latest balance

#### Which issue(s) does this PR fixes?:

Fixes #758

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: apps/web/src/components/BridgeForm.tsx [hunk 1]
```diff
@@ -130,28 +130,48 @@ export default function BridgeForm({
...
+    setTokenBalances(updatedBalances);
+    return updatedBalances;
+  }
+
+  const checkBalance = debounce(async () => {
+    getBalanceFn();
   }, 200);
 
-  useEffect(() => {
+  async function verifyTransfer(refetch?: boolean) {
```
Comment (pierregee) [inline apps/web/src/components/BridgeForm.tsx:151]:
```suggestion
  async function verifySufficientHWBalance(
    refetch?: boolean
  ): Promise<boolean | undefined> {
```
Is this func name better? verifyTransfer sounds a bit generic

File: apps/web/src/components/BridgeForm.tsx [hunk 2]
```diff
@@ -198,33 +218,37 @@ export default function BridgeForm({
   };
 
   const onTransferTokens = async (): Promise<void> => {
-    if (isSendingFromEthNetwork) {
-      // Revalidate entered amount after refetching EVM balance
-      const refetchedEvmBalance = await refetchEvmBalance();
-      if (
-        validateAmountInput(
+    setIsVerifyingTransaction(true);
+    const verify = await verifyTransfer(true);
```
Comment (pierregee) [inline apps/web/src/components/BridgeForm.tsx:222]:
```suggestion
    const isBalanceSufficient = await verifySufficientHWBalance(true);
    if (!isBalanceSufficient) {
       setIsVerifyingTransaction(false);
       return
    }
```

File: apps/web/src/components/BridgeForm.tsx [hunk 3]
```diff
@@ -130,28 +130,48 @@ export default function BridgeForm({
   const { getBalance } = useCheckBalance();
   const [isBalanceSufficient, setIsBalanceSufficient] = useState(true);
   const [tokenBalances, setTokenBalances] = useState({});
+  const [isVerifyingTransaction, setIsVerifyingTransaction] = useState(false);
 
-  const checkBalance = debounce(async () => {
+  async function getBalanceFn() {
```
Comment (pierregee) [inline apps/web/src/components/BridgeForm.tsx:135]: can add some typescript for the result, if possible

File: apps/web/src/components/BridgeForm.tsx [hunk 4]
```diff
@@ -130,28 +130,48 @@ export default function BridgeForm({
...
       [key]: balance,
-    });
+    };
+
+    setTokenBalances(updatedBalances);
+    return updatedBalances;
+  }
+
+  const checkBalance = debounce(async () => {
+    getBalanceFn();
```
Comment (pierregee) [inline apps/web/src/components/BridgeForm.tsx:148]: hmm could you also check why linting is not throwing error/warning on this?, promises should have await right?

File: apps/web/src/components/BridgeForm.tsx [hunk 5]
```diff
@@ -130,28 +130,48 @@ export default function BridgeForm({
...
   }, 200);
 
-  useEffect(() => {
+  async function verifyTransfer(refetch?: boolean) {
     const key = `${selectedNetworkA.name}-${selectedTokensA.tokenB.symbol}`;
-    const balance = tokenBalances[key];
+    const refetchBalance = refetch ? await getBalanceFn() : tokenBalances;
+    const balance = refetch ? refetchBalance[key] : tokenBalances[key];
```
Comment (pierregee) [inline apps/web/src/components/BridgeForm.tsx:154]:
```suggestion
   const balance = (refetch ? await getBalanceFn() : tokenBalances)[key];
```
hmm is this better? I noticed that the second ternary operator isn't necessary

**Review by pierregee — CHANGES_REQUESTED** [review]

File: apps/web/src/components/BridgeForm.tsx (hunk 5 above)
Comment (G9000) [reply apps/web/src/components/BridgeForm.tsx:154]: haha preference :v :v i guess

File: apps/web/src/components/BridgeForm.tsx (hunk 2 above)
Comment (G9000) [reply apps/web/src/components/BridgeForm.tsx:222]: already declared on top. So instead i change to isBalanceSufficientVerified?

File: apps/web/src/components/BridgeForm.tsx [hunk 6]
```diff
@@ -198,33 +225,37 @@ export default function BridgeForm({
...
+      if (isSendingFromEthNetwork) {
+        // Revalidate entered amount after refetching EVM balance
+        const refetchedEvmBalance = await refetchEvmBalance();
+        if (
+          validateAmountInput(
+            amount,
+            new BigNumber(refetchedEvmBalance.data?.formatted ?? 0)
+          )
+        ) {
+          return;
```
Comment (pierregee) [inline apps/web/src/components/BridgeForm.tsx:240]: If this if condition is satisfied, button will not be enabled again
```suggestion
          setIsVerifyingTransaction(false);
          return;
```

**Review by pierregee — CHANGES_REQUESTED** [review]

**Comment by JoshuaC817** [comment]
LGTM!

**Review by pierregee — APPROVED** [review]

**Pull request merged**